### ISSUE TITLE
lucky watch -r not work with Firefox (current used version is 110)

### ISSUE THREAD

**Issue by zw963** [luckyframework/lucky#1802]
**Describe the bug**
the following default web config

```yaml
web: lucky watch --reload-browser
```

Not work with Firefox, when websocket reload the browser, get a error page (check following screenshot)

![image](https://user-images.githubusercontent.com/549126/231514441-88591468-2b05-42f5-893e-6f21ecf53a28.png)

**Expected behavior**
It should works as Chrome with `lucky watch`

**Versions (please complete the following information):**
 - Lucky version 1.0.0
 - Crystal version 1.8.0
 - OS: Arch Linux

**Additional context**

One more issue.

I try with `lucky watch --reload-browser -w browsersync` too.

on my laptop, it not work with both firefox or chrome, it's seem like get blocked when reloading browser.

```
web          | $ /clover/node_modules/.bin/browser-sync reload --port 3001
web          | Done in 0.64s.
web          |  Done  compiling
web          | [Browsersync] Reloading Browsers...
```

**Comment by mdwagner** [comment]
Is that port open on your machine?

**Comment by zw963** [comment]
> Is that port open on your machine?

definitely

**Comment by mdwagner** [comment]
> it not work with both firefox or chrome, it's seem like get blocked when reloading browser.

So, this isn't Firefox specific, like mentioned in the title?

**Comment by zw963** [comment]
this exactly issue only happen on Firefox.

**Comment by mdwagner** [comment]
I wasn't able to reproduce this on Pop!_OS 22.04 for MS Edge (112.0.1722.64) and Firefox (112.0.1), but Google Chrome (112.0.5615.165) I was able to reproduce. Although, in Google Chrome's case, it refreshed on it's own even though it failed to load the page the first time.

My guess is, this is a race condition between when the server is stopped so it can recompile, and when it is officially back _up_. I suppose we could add a small delay before we tell the browser to reload, but I'm not sure there's much else we can do.

**Comment by zw963** [comment]
> My guess is, this is a race condition between when the server is stopped (so it can recompile), and when it is officially back up. I suppose we could add a small delay before we tell the browser to reload, but I'm not sure there's much else we can do.

Make sense to me.

**Comment by mdwagner** [comment]
Alright, since one of my machines recently has been having this exact problem (with MS Edge instead of Firefox), I decided to take another look at how to solve this.

I wasn't able to make it so the _server_ could delay when it tells the _browser_ when to reload. But I've found that if the _browser_ is able to delay the reload, this seems to solve it.

So, my current solution is to add a setTimeout for _1000 ms_ in the `live_reload_connect_tag` method, maybe even making it configurable in your `watch.yml`. Ideally, 1 second should be enough, but maybe depending on your hardware it sometimes isn't.
